src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -611,35 +649,44 @@ private static void RunTests(bool persistLogFileToDisk)
                 Verbose = false,
             };
 
-            // The exception raising rule produces a location free warning for every analysis target
-            int baseWarningsCount = files.Count();
+            // The exception raising rule produces a location free warning for every analysis target in cases
+            // where the exception raising rule is not, in fact, raising exceptions. Otherwise, the exception
+            // raising rule will produce one of a range of notifications. For notifications like 'pdb load 
+            // error', there will still be a notification raised for each scan target (but no per-file result).
+            int baseWarningsCount = raiseNotificationPerBinary ? 0 : files.Count();
 
             // When we run without caching results by hash, we should have a result for 
             // every file that produces a failure level of some kind (error & warning only since not running verbose).
             int expectedResultsCount = GetNonVerboseFailuresCount(files) + baseWarningsCount;
 
-            IList<Result> results = RunTest(mockFileSystem, options, expectedResultsCount);
+            Run run = RunTest(mockFileSystem, options, expectedResultsCount, expectedReturnCode);
+            IList<Result> results = run.Results;
```

Comment:
>results [](start = 26, length = 7)

BUG: We aren't actually verifying that the `results` and `notification` list have the right numbers of elements. We're just extracting them and overwriting them on the next test case.

Reply:
Yep, good eye, I meant to note this detail when I described these tests as a disaster. That lack of validation is an existing test bug. Will address this in the next rewrite.

---
In reply to: [316907814](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316907814) [](ancestors = 316907814)